### PR TITLE
FB-961 Improve performance of CM_Site_Model

### PR DESCRIPTION
**Pull request by fvovan**

### CODE REVIEW

**Comment by fvovan** [comment]
It's the simplest thing that we may do now.
Idea is we don't need sorting of  30 sites every time.  If it doesn't help we can aggressively cache `$_siteList` and `$_siteListSorted` for 5-10 minutes (to not care of invalidation). @{c7e8ab9c-90bf-4796-bb88-9aa7c8150044,Reto Kaiser} wdyt?

**Comment by njam** [comment]
Add 2 lazy functions:
- _getSiteListSorted()
- _getSiteList() <- Try caching the full array in Memcache

**Comment by fvovan** [comment]
I would suggest trying to deploy it without caching for now. Got stuck on serialization. For some reason it loses `_modules` and `_themes` attributes when unserialized, and we can't quickly define `CM_Site_Abstract::unserialize()` (parent's method is final). @{c7e8ab9c-90bf-4796-bb88-9aa7c8150044,Reto Kaiser} wdyt?

File: library/CM/Site/SiteFactory.php [hunk 1]
```diff
@@ -5,23 +5,13 @@ class CM_Site_SiteFactory {
     /** @var CM_Site_Abstract[] */
```
Comment (njam) [inline library/CM/Site/SiteFactory.php]: |`line 5`|
|---|
Docu: can be null

**Comment by njam** [comment]
sgtm

**Comment by njam** [comment]
Small docu, otherwise lgtm.
Feel free to merge and release when adjusted

**Pull request merged**